synapse/storage/databases/main/user_directory.py

```diff
@@ -388,67 +388,23 @@ def _update_profile_in_user_dir_txn(txn):
             if isinstance(self.database_engine, PostgresEngine):
                 # We weight the localpart most highly, then display name and finally
                 # server name
-                if self.database_engine.can_native_upsert:
-                    sql = """
+                sql = """
                         INSERT INTO user_directory_search(user_id, vector)
                         VALUES (?,
                             setweight(to_tsvector('simple', ?), 'A')
                             || setweight(to_tsvector('simple', ?), 'D')
                             || setweight(to_tsvector('simple', COALESCE(?, '')), 'B')
                         ) ON CONFLICT (user_id) DO UPDATE SET vector=EXCLUDED.vector
                     """
-                    txn.execute(
-                        sql,
-                        (
-                            user_id,
-                            get_localpart_from_id(user_id),
-                            get_domain_from_id(user_id),
-                            display_name,
-                        ),
-                    )
-                else:
-                    # TODO: Remove this code after we've bumped the minimum version
-                    # of postgres to always support upserts, so we can get rid of
-                    # `new_entry` usage
```

Comment:
I'm trusting that our minimum now supports upserts, I haven't double checked that.

Reply:
We now always return true for postgres \o/

https://github.com/matrix-org/synapse/blob/329ef5c715d81b538e8b071de046c698a82eae10/synapse/storage/engines/postgres.py#L125-L129